src/test/java/com/conveyal/datatools/manager/jobs/AutoDeployFeedJobTest.java

```diff
@@ -0,0 +1,194 @@
+    private static FeedSource mockFeedSourceC;
+    private static FeedVersion feedVersionC;
+    private static FeedSource mockFeedSourceD;
+    private static Project projectD;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        DatatoolsTest.setUp();
+        Auth0Connection.setAuthDisabled(true);
+        LOG.info("{} setup", AutoDeployFeedJobTest.class.getSimpleName());
+        user = Auth0UserProfile.createTestAdminUser();
+
+        // used by multiple tests.
+        server = createOtpServer();
+        DeployJob.DeploySummary deploySummary = createDeploymentSummary(server.id);
+
+        projectA = createProject();
+        deploymentA = createDeployment(deploySummary, projectA.id);
+        mockFeedSourceA = createFeedSource("Mock feed source A", projectA.id);
+        TestUtils.createFeedVersion(mockFeedSourceA,
+            TestUtils.zipFolderFiles("fake-agency-expire-in-2099"));
+
+        projectB = createProject();
+        deploymentB = createDeployment(deploySummary, projectB.id);
+        mockFeedSourceB = createFeedSource("Mock feed source B", projectB.id);
+        TestUtils.createFeedVersion(mockFeedSourceB,
+            TestUtils.zipFolderFiles("fake-agency-with-only-calendar-expire-in-2099-with-failed-referential-integrity"));
+
+        projectC = createProject();
+        deploymentC = createDeployment(deploySummary, projectC.id);
+        mockFeedSourceC = createFeedSource("Mock feed source C", projectC.id);
+        feedVersionC = getFeedVersionFromGTFSFile(mockFeedSourceC,
+            TestUtils.zipFolderFiles("fake-agency-expire-in-2099"));
+        Persistence.feedVersions.create(feedVersionC);
+
+        projectD = createProject();
+        mockFeedSourceD = createFeedSource("Mock feed source D", projectD.id);
+        TestUtils.createFeedVersion(mockFeedSourceD,
+            TestUtils.zipFolderFiles("fake-agency-expire-in-2099"));
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        Auth0Connection.setAuthDisabled(Auth0Connection.getDefaultAuthDisabled());
+        server.delete();
+
+        projectA.delete();
+        mockFeedSourceA.delete();
+        deploymentA.delete();
+
+        projectB.delete();
+        mockFeedSourceB.delete();
+        deploymentB.delete();
+
+        projectC.delete();
+        mockFeedSourceC.delete();
+        deploymentC.delete();
+
+        projectD.delete();
+        mockFeedSourceD.delete();
+        Persistence.feedVersions.removeById(feedVersionC.id);
+    }
+
+    @Test
+    public void canAutoDeployFeedVersionForProject() throws IOException {
+        projectA.pinnedDeploymentId = deploymentA.id;
+        projectA.feedSources = new ArrayList<>();
+        projectA.feedSources.add(mockFeedSourceA);
+        Persistence.projects.replace(projectA.id, projectA);
+        AutoDeployJob autoDeployFeedJob = new AutoDeployJob(projectA, user);
+        autoDeployFeedJob.run();
+        assertNotNull(projectA.lastAutoDeploy);
```

Comment:
For each of these final `assert(Not)Null` checks, can you write a comment describing why they should (not) be null?

Reply:
Updated.